### ISSUE TITLE
[PyTorch] How to restore fp8 amp training from checkpoint

### ISSUE THREAD

**Issue by alexdremov** [NVIDIA/TransformerEngine#982]
Hello!

I try to implement a multi-stage training with fp8 autocast. However, when I load checkpoint from first training stage using torch's `load_state_dict(...)`, loss quickly explodes. 

Are there any global fp8 states that also need to be saved/restored? 

Optimizer's state and grad scaler state are restored. Such behavior is not reproduced with amp fp16 training

**Comment by alexdremov** [comment]
@timmoon10 are there any thoughts why training could be restored incorrectly?

**Comment by alexdremov** [comment]
> The FP8 scaling factors should be automatically included in the state dict since our modules implement [`get_extra_state`](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.get_extra_state)/[`set_extra_state`](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.get_extra_state). However, if you switch between FP8 and non-FP8 training, you may need to recalibrate the scaling factors to avoid numerical weirdness. Try passing `enabled=False` and `calibrating=True` to `fp8_autocast` and doing a forward and backward pass (see [here](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/pytorch.html#transformer_engine.pytorch.fp8_autocast)).

Yes, I've seen those methods implemented, but still it seems like something is not restored

**Comment by avizon-aws** [comment]
Where is the get_extra_state method invoked? Didnt see it being used anywhere, I do see the set_extra_state being invoked in _load_from_state_dict.

**Comment by sbhavani** [comment]
Closing as answered. FP8 scaling factors are automatically saved/restored.
